lua/cmp_luasnip/init.lua

```diff
@@ -137,19 +137,40 @@ function source:execute(completion_item, callback)
 	local cursor = vim.api.nvim_win_get_cursor(0)
 	-- get_cursor returns (1,0)-indexed position, clear_region expects (0,0)-indexed.
 	cursor[1] = cursor[1] - 1
+	local line = require("luasnip.util.util").get_current_line_to_cursor()
+
+	local expand_params = snip:matches(line)
```

Comment:
I don't think the snippet should expand if `matches` returns `nil`, since that generally means that some env-variables that the snippet depends on are not set.
For example, with regex-trigger, if the regex does not match, and the snippet assumes that it does, and that there is a `env.LS_CAPTURE_1`, this would produce an error.

Unfortunately, that isn't something we can circumvent, if the pattern does not match, the user has to provide text that does, we essentially can't and have to reject the expansion :/ (or, in the case of lua-patterns, expand the expand-helper, see a few lines above).

Reply:
if `snip:matches` returns nil, this patch just do nothing. so im not sure what the worry about..

Reply:
Ah, true!
Okay, I guess we can tackle this some other time

Reply:
@L3MON4D3 happy to merge?

Reply:
Yup, I think this should be fine.
Nice work @woodgear 👍